pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go

```diff
@@ -98,3 +120,77 @@ func TestGetBranches(t *testing.T) {
 		})
 	}
 }
+
+/*
+ * Testing for getMergeRequests
+ */
+func TestGetMergeRequests(t *testing.T) {
+	testCases := []struct {
+		desc            string
+		client          graphql.Client
+		expectedErr     error
+		expectedNodeLen int
+	}{
+		{
+			desc:            "empty mergeRequestData",
+			client:          &mockClient{},
+			expectedErr:     nil,
+			expectedNodeLen: 0,
+		},
+		{
+			desc:        "produce error in client",
+			client:      &mockClient{err: true, errString: "An error has occurred"},
+			expectedErr: errors.New("An error has occurred"),
+		},
+		{
+			desc: "valid mergeRequestData",
+			client: &mockClient{
+				maxPages: 1,
```

Comment:
You could have another test for multiple pages and no pages

Reply:
Good idea. Thank!